File: Python/app/db/models/vehiculo.py
from sqlalchemy import Column, Integer, String
from app.db.base import Base

class Vehiculo(Base):
    __tablename__ = "vehiculo"
    id_vehiculo = Column(Integer, primary_key=True, index=True)
    placa = Column(String, unique=True, index=True, nullable=False)
    marca = Column(String, nullable=False)
    modelo = Column(String, nullable=False)
    anio = Column(Integer, nullable=False)
    tipo_id = Column(String, nullable=False)  # Simulado como texto
    estado = Column(String, nullable=False)

File: modules/modulo_administracion_vehiculos/src/main/java/com/asw/modulo_administracion_vehiculos/model/entity/Vehiculo.java
package com.asw.modulo_administracion_vehiculos.model.entity;


import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = "vehiculos")
public class Vehiculo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_vehiculo")
    private Long idVehiculo;

    private String placa;
    private String marca;
    private String modelo;
    private Integer anio;
    private String estado;

    @ManyToOne
    @JoinColumn(name = "tipo_id")
    private TipoVehiculo tipo;
}